### ISSUE TITLE
Service provider exception when dumping the CFG 

### ISSUE THREAD

**Issue by gigiblender** [oracle/graal#3303]
**Describe the issue**

Running `$JAVA_HOME/bin/java -XX:-TieredCompilation -Xcomp -Dgraal.Dump=*:1 -Dgraal.PrintCFG=true -XX:CompileOnly=Main::testMethod Main` with `JAVA_HOME` pointing to GraalVM CE 21.0.0.2 will dump the CFG up to the `PostAllocationOptimizationStage`. The `code generation` and `code installation` phases will not appear in the dump. 

![image](https://user-images.githubusercontent.com/7903443/112050464-cab99c00-8b48-11eb-8a4c-428a1d9ef8d3.png)

The stack trace below is generated: 
```
CFGPrinter: Exception during output of After code generation: java.lang.InternalError: No org.graalvm.compiler.code.DisassemblerProvider providers found when building native image
java.lang.InternalError: No org.graalvm.compiler.code.DisassemblerProvider providers found when building native image
	at org.graalvm.compiler.serviceprovider.GraalServices.load(GraalServices.java:96)
	at org.graalvm.compiler.printer.CFGPrinterObserver.selectDisassemblerProvider(CFGPrinterObserver.java:263)
	at org.graalvm.compiler.printer.CFGPrinterObserver.disassemble(CFGPrinterObserver.java:286)
	at org.graalvm.compiler.printer.CFGPrinterObserver.dumpSandboxed(CFGPrinterObserver.java:227)
	at org.graalvm.compiler.printer.CFGPrinterObserver.dump(CFGPrinterObserver.java:85)
	at org.graalvm.compiler.debug.ScopeImpl.dump(ScopeImpl.java:223)
	at org.graalvm.compiler.debug.DebugContext.dump(DebugContext.java:1357)
	at org.graalvm.compiler.core.gen.LIRCompilerBackend.emitCode(LIRCompilerBackend.java:248)
	at org.graalvm.compiler.core.gen.LIRCompilerBackend.emitBackEnd(LIRCompilerBackend.java:88)
	at org.graalvm.compiler.core.target.Backend.emitBackEnd(Backend.java:279)
	at org.graalvm.compiler.core.GraalCompiler.compile(GraalCompiler.java:146)
	at org.graalvm.compiler.core.GraalCompiler.compileGraph(GraalCompiler.java:130)
	at org.graalvm.compiler.hotspot.HotSpotGraalCompiler.compileHelper(HotSpotGraalCompiler.java:238)
	at org.graalvm.compiler.hotspot.HotSpotGraalCompiler.compile(HotSpotGraalCompiler.java:256)
	at org.graalvm.compiler.hotspot.CompilationTask$HotSpotCompilationWrapper.performCompilation(CompilationTask.java:172)
	at org.graalvm.compiler.hotspot.CompilationTask$HotSpotCompilationWrapper.performCompilation(CompilationTask.java:84)
	at org.graalvm.compiler.core.CompilationWrapper.run(CompilationWrapper.java:222)
	at org.graalvm.compiler.hotspot.CompilationTask.runCompilation(CompilationTask.java:336)
	at org.graalvm.compiler.hotspot.HotSpotGraalCompiler.compileMethod(HotSpotGraalCompiler.java:154)
	at org.graalvm.compiler.hotspot.HotSpotGraalCompiler.compileMethod(HotSpotGraalCompiler.java:637)
	at org.graalvm.compiler.hotspot.HotSpotGraalCompiler.compileMethod(HotSpotGraalCompiler.java:110)
	at jdk.vm.ci.hotspot.HotSpotJVMCIRuntime.compileMethod(HotSpotJVMCIRuntime.java:807)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_VA_LIST_Nonvirtual:Ljdk_vm_ci_hotspot_HotSpotJVMCIRuntime_2_0002ecompileMethod_00028Ljdk_vm_ci_hotspot_HotSpotResolvedJavaMethod_2IJI_00029Ljdk_vm_ci_hotspot_HotSpotCompilationRequestResult_2(JNIJavaCallWrappers.java:0)
```

Not using libgraal ( `-XX:-UseJVMCINativeLibrary`) solves the problem.

**Comment by oubidar-Abderrahim** [comment]
Thank you @gigiblender for reporting this.

@dougxc Could you please advise on this issue?

**Comment by dougxc** [comment]
There is a simple fix for this:
```
diff --git a/substratevm/src/com.oracle.svm.graal.hotspot.libgraal/src/com/oracle/svm/graal/hotspot/libgraal/LibGraalFeature.java b/substratevm/src/com.oracle.svm.graal.hotspot.libgraal/src/com/oracle/svm/graal/hotspot/libgraal/LibGraalFeature.java
index cab2067da1b..85dbebb18ac 100644
--- a/substratevm/src/com.oracle.svm.graal.hotspot.libgraal/src/com/oracle/svm/graal/hotspot/libgraal/LibGraalFeature.java
+++ b/substratevm/src/com.oracle.svm.graal.hotspot.libgraal/src/com/oracle/svm/graal/hotspot/libgraal/LibGraalFeature.java
@@ -56,6 +56,7 @@ import java.util.stream.Collectors;
 import org.graalvm.collections.EconomicMap;
 import org.graalvm.collections.MapCursor;
 import org.graalvm.compiler.api.replacements.SnippetReflectionProvider;
+import org.graalvm.compiler.code.DisassemblerProvider;
 import org.graalvm.compiler.core.GraalServiceThread;
 import org.graalvm.compiler.debug.DebugContext;
 import org.graalvm.compiler.debug.DebugHandlersFactory;
@@ -446,6 +447,7 @@ public final class LibGraalFeature implements com.oracle.svm.core.graal.GraalFea
         GraalServices.load(PartialEvaluatorConfiguration.class);
         GraalServices.load(HotSpotCodeCacheListener.class);
         GraalServices.load(HotSpotMBeanOperationProvider.class);
+        GraalServices.load(DisassemblerProvider.class);

         try (DebugContext.Scope scope = debug.scope("SnippetSupportEncode")) {
             InvocationPlugins compilerPlugins = hotSpotSubstrateReplacements.getGraphBuilderPlugins().getInvocationPlugins();
```
@oubidar-Abderrahim could you please help merge that fix.

**Comment by oubidar-Abderrahim** [comment]
fix merged in 28e94a58852343644ee94922c0a3440f8fcd2b56

**Comment by oubidar-Abderrahim** [comment]
If the error still persists after the fix, please reopen this issue.
